src/rgw/rgw_http_client.cc

```diff
@@ -790,7 +803,14 @@ void *RGWHTTPManager::reqs_thread_entry()
     }
   }
 
+
   RWLock::WLocker rl(reqs_lock);
+  for (auto r : unregistered_reqs) {
+    _finish_request(r, -ECANCELED);
+  }
+
+  unregistered_reqs.clear();
+
```

Comment:
ok, i can see how these could leak since the last call to `manage_pending_requests()`. i was worried that we might still leak them after this point, but after further investigation i'm not seeing how that would happen